### ISSUE TITLE
[Blazor] Browser scroll is not set to top page on navigation

### ISSUE THREAD

**Issue by glararan** [aspnet/AspNetCore#14471]
### Describe the bug
Upgrading from preview-8 to GA caused browser is no longer reseted to top of the page on navigation.

### To Reproduce
Currently I dont know how to reproduce. New project work, then I simply broke it somehow, any idea how to debug it?

### Expected behavior
Reset navigation to scrollY = 0

### Additional context
.NET Core SDK (reflecting any global.json):
 Version:   3.0.100
 Commit:    04339c3a26

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17134
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.0.100\

Host (useful for support):
  Version: 3.0.0
  Commit:  7d57652f33

```
<PackageReference Include="BlazorInputFile" Version="0.1.0-preview" />
    <PackageReference Include="ChartJs.Blazor.Fork" Version="0.10.3-preview" />
    <PackageReference Include="CurrieTechnologies.Razor.SweetAlert2" Version="1.0.0" />
    <PackageReference Include="Google.Apis.Calendar.v3" Version="1.41.1.1713" />
    <PackageReference Include="HtmlAgilityPack" Version="1.11.12" />
    <PackageReference Include="Magick.NET-Q16-AnyCPU" Version="7.14.4" />
    <PackageReference Include="MailKit" Version="2.3.1.6" />
    <PackageReference Include="Microsoft.AspNet.Identity.Core" Version="2.2.2" />
    <PackageReference Include="Microsoft.AspNetCore.Authentication.Facebook" Version="3.0.0" />
    <PackageReference Include="Microsoft.AspNetCore.Authentication.Google" Version="3.0.0" />
    <PackageReference Include="Microsoft.AspNetCore.Authentication.Twitter" Version="3.0.0" />
    <PackageReference Include="Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore" Version="3.0.0" />
    <PackageReference Include="Microsoft.AspNetCore.Identity.EntityFrameworkCore" Version="3.0.0" />
    <PackageReference Include="Microsoft.AspNetCore.Identity.UI" Version="3.0.0" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="3.0.0" />
    <PackageReference Include="Microsoft.EntityFrameworkCore.Tools" Version="3.0.0">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
    <PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="3.0.0" />
    <PackageReference Include="Microsoft.VisualStudio.Web.CodeGeneration.Design" Version="3.0.0" />
    <PackageReference Include="System.Data.SqlClient" Version="4.7.0" />
    <PackageReference Include="System.Drawing.Common" Version="4.6.0" />
```

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us, @glararan.
Unfortunately we can't reproduce this and we will need a minimalistic repro project (a GitHub repo will be handy) to investigate.

**Comment by glararan** [comment]
@mkArtakMSFT Ok I created minimalistic repro of huge project. That minimalistic repro contains still bug. Is it possible to send it privately? Do you need database aswell?

**Comment by mkArtakMSFT** [comment]
@glararan we are a small team and don't have capacity to debug actual projects.
The reported concern on this issue is scrolling, which shouldn't involve database at all. The ideal repro project is a new project with only the minimum required  content resulting in the issue to reproduce.

**Comment by glararan** [comment]
@mkArtakMSFT Ok, my apologies. I figured out which part is causing issue. Still cant figure out why.

I have `MainLayout `containing component `<BusBar></BusBar>`
with following code

`@inherits OwningComponentBase<myService>

<div class="card rounded-0" id="bus-bar">
    <div class="card-header" id="bus-bar-head" data-toggle="collapse" data-target="#bus-bar-content" aria-expanded="true" aria-controls="bus-bar-content">Title</div>

    <div id="bus-bar-content" class="collapse show" aria-labelledby="bus-bar-head">
        <div class="card-body">
            <div id="a-b">
                <h3>A</h3>

                <Loader Object="ToA">
                    @foreach (var bus in ToA.Take(3))
                    {
                    <Bus Item="@bus" OnFinished="@((BusEventArgs __e) => RemoveBus(ToA, __e.Bus))"></Bus>
                    }
                </Loader>
            </div>

            <div id="b-a">
                <h3>B</h3>

                <Loader Object="ToB">
                    @foreach (var bus in ToB.Take(3))
                    {
                    <Bus Item="@bus" OnFinished="@((BusEventArgs __e) => RemoveBus(ToB, __e.Bus))"></Bus>
                    }
                </Loader>
            </div>
        </div>
    </div>
</div>

@code
{
    public List<Models.Bus> ToA;
    public List<Models.Bus> ToB;

    void RemoveBus(List<Models.Bus> list, Models.Bus bus)
    {
        list.Remove(bus);

        if (list.Count < 3)
        {
            BusesModel buses = Service.GetBuses();

            ToA= buses.ToA;
            ToB= buses.ToB;
        }
    }

    protected override void OnInitialized()
    {
        BusesModel buses = Service.GetBuses();

        ToA = buses.ToA;
        ToB = buses.ToB;
    }
}`

Component <Bus>
```
@using System.Threading;

<div class="time-schedule">
    @(Item.Today ? "Today" : "Tomorrow") <b>@Date.ToString("HH:mm")</b> - BUS <b>@Item.BusNumber</b> <b>@(Hours > 0 ? $"{Hours}h ": "")@(Minutes)m @(Seconds)s</b>
</div>

@code
{
[Parameter]
public Models.Bus Item { get; set; }

[Parameter]
public EventCallback<BusEventArgs> OnFinished { get; set; }

public bool Today { get; set; }
public DateTime Date => DateTime.Parse(Item?.Time);

public int Hours { get; set; }
public int Minutes { get; set; }
public int Seconds { get; set; }

protected override void OnInitialized()
{
    Timer timer = new Timer(new TimerCallback(_ =>
    {
        Seconds = (int)(Date - DateTime.Now).TotalSeconds % 60;
        Minutes = (int)(Date - DateTime.Now).TotalMinutes % 60;
        Hours = (int)(Date - DateTime.Now).TotalHours;

        InvokeAsync(() => StateHasChanged());

        if(Seconds == 0 && Minutes == 0 && Hours == 0)
            InvokeAsync(async() => await OnFinished.InvokeAsync(new BusEventArgs(Item)));
    }), null, 0, 1000);
}
}
```


Rest of needed code:
`    public class BusEventArgs : EventArgs
    {
        public Bus Bus { get; private set; }

        public BusEventArgs(Bus bus)
        {
            Bus = bus;
        }
    }

    public class Bus
    {
        public string Day { get; set; }
        public string Time { get; set; }
        public string BusNumber { get; set; }
        public bool Today { get; set; }
    }

    public class BusesModel
    {
        public List<Bus> ToDejvicka { get; set; }
        public List<Bus> ToSuchdol { get; set; }
    }`


Loader component (there is no issue at all just to give u all codes) 
```
@if (Object is null)
{
<div class="col-12 my-5">
    <div class="d-flex justify-content-center">
        <div class="spinner-border text-success" role="status">
            <span class="sr-only">Loading...</span>
        </div>
    </div>
</div>
}
else
{
@ChildContent
}

@code
{
[Parameter]
public object Object { get; set; }

[Parameter]
public RenderFragment ChildContent { get; set; }
}

```

**Comment by mkArtakMSFT** [comment]
This looks a bit better. We're still not there yet. I think by trimming your code further to find the actual culprit in this code will help you too find out what is causing this issue. Please try to remove as much code as possible, until you get to the actual line, removal of which will not reproduce the behavior any more.

**Comment by glararan** [comment]
I did not solve it yet. But after removing BusBar component I still have issue on few pages. So I have to investigate futher. However I found out strange behavior! Moving from page D to page E then back to page D clicking like 30 times go to E, going back to D, going to E on 30rd click approximately I got browser.scrollY to 0. What the heck? What could trigger on 30rd(+-) click to make it reset? I was just clicking on <NavLink> buttons.

**Comment by Andrzej-W** [comment]
I don't know if this is related to your problem, but this code is wrong:
```c#
Seconds = (int)(Date - DateTime.Now).TotalSeconds % 60;
Minutes = (int)(Date - DateTime.Now).TotalMinutes % 60;
Hours = (int)(Date - DateTime.Now).TotalHours;
```
You read current time 3 times. It is possible that when you read it first time it is 10:59:59 and when you read it second and third time it is 11:00:00. Maybe this bug ruins your program logic in some other places, not only here:
```c#
if(Seconds == 0 && Minutes == 0 && Hours == 0)
            InvokeAsync(async() => await OnFinished.InvokeAsync(new BusEventArgs(Item)));
    }), null, 0, 1000);
```

**Comment by glararan** [comment]
@Andrzej-W I dont think that's source of issue. Anyway I will change it to read it once. As I see it it should read it in few nanosecs.

**Comment by glararan** [comment]
@mkArtakMSFT I figured out what is causing this bug! Its not BusBar as I thought. I just had that random luck that one of many times it worked. 

So this bug.. how to reproduce it:

Implement custom javascript file e.g. core.js, add it after Blazor <script> with following code:
```
window.scrollTo = function (element)
{
    if ($(element).length < 1)
        return;

    $("html, body").animate({ scrollTop: $(element).offset().top });
};
```

Is **scrollTo** reserved name? I changed it for few attempts to **scrollToElement** and voala.. bug is gone
